cmd/fluxctl/install_cmd.go

```diff
@@ -28,74 +22,32 @@ func newInstall() *installOpts {
 func (opts *installOpts) Command() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "install",
-		Short: "Bootstrap Flux, installing it in the cluster and initializing its manifests in the Git repository",
-		RunE:  opts.RunE,
+		Short: "Print and tweak Kubernetes manifests needed to install Flux in a Cluster",
+		Example: `# Install Flux and make it use Git repository git@github.com:<your username>/flux-get-started
+fluxctl install --git-url 'git@github.com:<your username>/flux-get-started' | kubectl -f -`,
+		RunE: opts.RunE,
 	}
-	cmd.Flags().StringVarP(&opts.gitURL, "git-url", "r", "",
+	cmd.Flags().StringVarP(&opts.gitURL, "git-url", "", "",
 		"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
-	cmd.Flags().StringVarP(&opts.gitBranch, "git-branch", "b", "master",
+	cmd.Flags().StringVarP(&opts.gitBranch, "git-branch", "", "master",
 		"Git branch to be used by Flux")
-	cmd.Flags().StringVarP(&opts.gitLabel, "git-label", "l", "flux",
-		"Git label to keep track of FLux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
-	cmd.Flags().StringVarP(&opts.gitFluxPath, "git-flux-subdir", "p", "flux/",
-		"Directory within the Git repository where to commit the Flux manifests")
-	cmd.Flags().DurationVarP(&opts.timeout, "timeout", "t", 10*time.Second,
-		"Timeout duration for I/O operations")
-	cmd.Flags().StringVarP(&opts.namespace, "namespace", "n", "flux",
+	cmd.Flags().StringVarP(&opts.gitLabel, "git-label", "", "flux",
+		"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+	cmd.Flags().StringVarP(&opts.gitUser, "git-user", "", "Flux",
+		"Username to use as git committer")
+	cmd.Flags().StringVarP(&opts.gitEmail, "git-email", "", "flux-dev@googlegroups.com",
+		"Email to use as git committer")
+	cmd.Flags().StringVarP(&opts.namespace, "namespace", "", "flux",
 		"Cluster namespace where to install flux")
-	cmd.Flags().BoolVarP(&opts.amend, "amend", "a", false,
-		"Stop for amending the Flux manifests before pushing them to the Git repository")
-	cmd.Flags().StringSliceVarP(&opts.additionalFluxArgs, "extra-flux-args", "e", []string{},
+	cmd.Flags().StringSliceVarP(&opts.additionalFluxArgs, "extra-flux-args", "", []string{},
```

Comment:
Remove this. People can edit the output as well as they can go find the option they want and encode here.

Reply:
Is it OK if we keep the functionality internally? (i.e. not exposed in a flag but accessible programmatically through a Go import). I am planning to use this in `eksctl`.

Reply:
To be frank I think I would find it super useful as a user. Flux arguments are the one thing which users will want to change all the time but it's not worth being exhaustive with all the options offered (i.e. I think that `--extra-flux-args` are a good compromise).

Reply:
Yes to that first question.

On the second comment, I suppose you might want to script the generation of flux configs, and for that an escape would be more useful (because you'd figure out the value for it once, and run it many times).

For interactive use, it would surely be easier to generate the file, then edit it and figure out which things you need to add and update.

Anyway: the bit you need is to have the escape available in the code, so neither of us have to be right or wrong here :-)